app/views/shared/_tree.html.haml

```diff
@@ -1,5 +1,3 @@
-- angular ||= false
-
-%div{:id => "#{name}_div"}
-  .treeview-pf-hover.treeview-pf-select{:id => "#{name}box", :style => "overflow: hidden"}
-    = render :partial  => "layouts/tree", :locals => tree.locals_for_render.merge(:angular => angular)
+.treeview{:id => "#{tree.name}box"}
```

Comment:
This no longer creates the `whatever_tree_div`, only `whatever_treebox`.

But we still have logic to replace tree divs (in ops diagnostics).

Keep the div too?

Reply:
Some divs are replaced via `_treebox` some via `_tee_div` so I'll just create a separate PR where I replace the `_tree_div` changes with `_treebox`.

Reply:
Wait, no, these two trees in diagnostics aren't right-side trees and so not affected by these changes.

Reply:
So for me, who is still a newbie in this. This is okey, we need only `_treebox` and therefore no change is needed?

Reply:
You don't need to do any changes here.

Reply:
...wait, are there other trees not rendered through these partials?

This completely removes the only places doing `{tree}_div`.
So if you're saying this doesn't convert all the trees, I need to re-review :D

Reply:
Also, if you're saying that, I'm really sad, because this layout was always supposed to be *the* only place to render trees :(.

Reply:
But good point, `miqInitTree` no longer exists, which probably means all the old code should be obsolete too, right?

Reply:
There are trees rendered via angular on the left side of explorer screens. That was from the time when we started converting stuff to angular and then it got stuck on accordion bugs from the PF side.

What do you mean all the old code? There are still some stuff the left-side trees use from `miq_tree.js`, that's why @brumik is calling the `isRedux` on the functions that are called from both the left-side and the right-side trees.

Reply:
Aaah.. so, this does convert all the trees, except for Generic Object?

Maybe we could just drop the `isRedux` logic and convert the tree?

Reply:
No, all the trees on the **right side**, not the trees in explorer's left side accordions.

Reply:
But also back to the original point .. @brumik please add the `_div`, we're still using it...

```
app/assets/javascripts/miq_application.js
56:  miqDimDiv(tree + '_div', true);

app/controllers/application_controller/advanced_search.rb
171:      page.replace("#{tree_name}_div",  :partial => "shared/tree",               :locals => {:tree => tree, :name => tree_name})

app/controllers/ops_controller/diagnostics.rb
658:        page.replace("settings_tree_div", :partial => "tree", :locals => {:name => "settings_tree"})
659:        page.replace("diagnostics_tree_div", :partial => "tree", :locals => {:name => "diagnostics_tree"})
665:        page.replace("zone_tree_div", :partial => "zone_tree")
```

Reply:
@skateman Do you have a list? Something to search for?

Reply:
https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/presenters/tree_builder.rb#L292

Reply:
Anything in this structure is rendered through angular, not through the shared/tree...and this PR doesn't cover them.

Reply:
I was looking for an example of one of the angular trees being used somewhere.
Trying to see how many it is.

Reply:
probably this:..

```
app/views/layouts/listnav/_generic_object_definition_show_list_with_treeview.html.haml
4:    ManageIQ.tree.data['#{@tree.name}'] = #{@tree.locals_for_render[:bs_tree]};
8:                   :reselect   => @tree.locals_for_render[:allow_reselect],
9:                   "ng-init"   => "vm.initData('#{@tree.name}', '#{@tree.locals_for_render[:select_node]}')",

app/views/layouts/listnav/_explorer.html.haml
10:          ManageIQ.tree.data['#{tree.name}'] = #{tree.locals_for_render[:bs_tree]};
13:                       :reselect   => tree.locals_for_render[:allow_reselect],
14:                       "ng-init"   => "vm.initData('#{tree.name}', '#{tree.locals_for_render[:select_node]}')",

app/views/shared/_tree.html.haml
2:  = react('HierarchicalTreeView', tree.locals_for_render)

app/views/miq_policy/_alert_profile_assign.html.haml
38:      = react('HierarchicalTreeView', @assign[:obj_tree].locals_for_render)
```

So...
we currently have 4 ways of using TreeBuilder trees..

left side:
* angular trees (listnav/_explorer)
* angular generic object tree

right side:
* react trees
* react alert profile assign tree

Is there an issue for this? Can you create one?

And, is there any reason for the 1 extra react tree?

Reply:
Way 3 and way 4 are the same ones, @brumik just didn't unify them. But there's also another one for automate embedded methods and which is the same as the dialog editor.

Reply:
The takeaway from this for @brumik is to make `app/views/miq_policy/_alert_profile_assign.html.haml` use the `shared/tree` like before :wink: